service/dap/server.go

```diff
@@ -433,7 +464,9 @@ func (s *Server) handleRequest(request dap.Message) {
 func (s *Server) send(message dap.Message) {
 	jsonmsg, _ := json.Marshal(message)
 	s.log.Debug("[-> to client]", string(jsonmsg))
+	s.sending.Lock()
 	_ = dap.WriteProtocolMessage(s.conn, message)
+	s.sending.Unlock()
```

Comment:
I went with simple lock here like in the rpc server. I think I need to tweak things a bit more to make sure it doesn't conflict with mu (e.g. in Stop()). An alternative would be to use a channel for all the sends and to have a dedicated goroutine that reads from that channel and sends over the connection. WDYT?

Reply:
This will prevent any accidental lock conflicts and also protect us from very slow network sends.